nengo/tests/test_solvers.py

```diff
@@ -624,6 +619,56 @@ def test_non_compositional_solver(Simulator, solver, rng, seed, plt):
     plt.plot(sim.trange(), y)
 
     assert np.allclose(y, -x, atol=0.1)
+    if scipy_sparse is not None:
+        assert isinstance(sim.data[conn].weights, scipy_sparse.csr_matrix)
+
+
+@pytest.mark.parametrize('solver_cls', [
+    LstsqDrop,
+    LstsqL1,
+])
+def test_sparse_decoders_warnings(Simulator, solver_cls):
+    if solver_cls is LstsqL1:
+        pytest.importorskip('sklearn')
+
+    # sparse decoders are not yet supported
+    solver = solver_cls()
+    assert solver.sparse
+    assert not solver.weights
+
+    with nengo.Network() as net:
+        conn = nengo.Connection(
+            nengo.Ensemble(10, 1), nengo.Node(size_in=1), solver=solver)
+
+    with pytest.warns(UserWarning, match="Sparse decoders"):
+        with Simulator(net) as sim:
+            pass
+
+    assert 0 < sim.data[conn].solver_info['sparsity'] < 1
+
+
+@pytest.mark.parametrize('solver_cls', [
+    LstsqDrop,
+    LstsqL1,
+])
+def test_sparse_without_scipy(Simulator, solver_cls):
+    if scipy_sparse is not None:
+        pytest.skip("Test requires no Scipy")
+
+    if solver_cls is LstsqL1:
+        pytest.importorskip('sklearn')
```

Comment:
Since `sklearn` requires `scipy`, this test doesn't actually do anything in this case. Test only makes sense for `LstsqDrop`.